Summarize current interpretation of the NeXus format (#262)
scipp/scippneutron: src/scippneutron/nexus/docs/our-interpretation-of-the-nexus-format.md

1. *Both* approaches are permitted.
2. For a concrete group in a NH there must not be more than one `NXdata` or `NXevent_data` child (or field thereof) within the group.
   If there is more than one, the group is considered invalid.
3. Certain classes pre-define aspects that also `NXdata` could deliver.
   For example, `NXdata` uses the [`signal` attribute](https://manual.nexusformat.org/classes/base_classes/NXdata.html#nxdata-signal-attribute) to define the field providing the signal array.
   For `NXdetector` the NF pre-defines that its signal is `data`, so if that is found it will be used, even if not `signal` attribute is present.

More concretely this means that, e.g., for loading an `NXdetector` from a NH, the implementation will:

1. Find all `NXdata` children.
2. Find all `NXevent_data` children.
3. Search the group for fields defined in `NXdata`.
   This includes looking for the `signal` attributes on the group and on fields.
4. Search the group for fields defined in `NXevent_data`.
5. Search the group for fields pre-defined in the class that are equivalents of what is defined in `NXdata`, even if the `NXdata` requirements (such as `signal` attributes) are not met.

> If the above results in not more than one item, the group can be loaded.

Reviewer: ```suggestion
If the above yields no more than one item, the group can be loaded.
```

## Missing axis labels

[NX_data](https://manual.nexusformat.org/classes/base_classes/NX_data.html) uses the [`axes` attribute](https://manual.nexusformat.org/classes/base_classes/NXdata.html#nxdata-axes-attribute) to define the names of fields that store coordinates for axes.
There is a legacy mechanism where the signal field has an [`axes` attribute](https://manual.nexusformat.org/classes/base_classes/NXdata.html#nxdata-data-axes-attribute) and this should not be used according to the NF.

The `axes` attribute uses `'.'` to define an axis without coordinate field.
The implication of the above is that there is no way to define the *name* of an axis.
For example, an `NXdata` group storing a stack of images may not define a coordinate for the "image" dimension of the signal dataset.
The NF does not specify any other way to define such an axis name.
We therefore have to fall back to a generic and meaningless dimension label.

Note that the `axes` attribute of the signal field could in principle be used to define such names and the `axes` attribute of the group would then only define the field names of coordinates.
That is, both attributes would work in conjunction.

> It is unclear whether this is allowed or actually considered invalid in the NF.

Reviewer: Since you have something about missing axes, should we also add something about missing units?

Author: Actually I don't know if we have reached a conclusion on this one?
